docs/DEVELOPER_GUIDE.md

```diff
@@ -1 +1,179 @@
+
+### Install Knative on a Kubernetes cluster
+
+1. [Set up a kubernetes cluster and install Knative Serving](https://knative.dev/docs/install/)
+
+- **Note**: KFServing currently only requires `Knative Serving` for auto-scaling, canary rollout,
+ `Istio` for traffic routing and ingress. You can follow instructions on
+ [Custom Install](https://knative.dev/docs/install/knative-custom-install) to install `Istio` and `Knative Serving`,
+ observability plug-ins are good to have for monitoring.
+
+### Setup your environment
+
+To start your environment you'll need to set these environment variables (we
+recommend adding them to your `.bashrc`):
+
+1. `GOPATH`: If you don't have one, simply pick a directory and add
+   `export GOPATH=...`
+1. `$GOPATH/bin` on `PATH`: This is so that tooling installed via `go get` will
+   work properly.
+1. `KO_DOCKER_REPO`: The docker repository to which developer images should be
+   pushed (e.g. `docker.io/<username>/[project]`).
+
+- **Note**: Set up a docker repository for pushing images. You can use any container image registry by adjusting 
+the authentication methods and repository paths mentioned in the sections below.
+   - [Google Container Registry quickstart](https://cloud.google.com/container-registry/docs/pushing-and-pulling)
+   - [Docker Hub quickstart](https://docs.docker.com/docker-hub/)
+   - [Azure Container Registry quickstart](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal)
+- **Note**: if you are using docker hub to store your images your
+  `KO_DOCKER_REPO` variable should be `docker.io/<username>`.
+- **Note**: Currently Docker Hub doesn't let you create subdirs under your
+  username.
+
+`.bashrc` example:
+
+```shell
+export GOPATH="$HOME/go"
+export PATH="${PATH}:${GOPATH}/bin"
+export KO_DOCKER_REPO='docker.io/<username>'
+```
+
+### Checkout your fork
+
+The Go tools require that you clone the repository to the
+`src/github.com/kubeflow/kfserving` directory in your
+[`GOPATH`](https://github.com/golang/go/wiki/SettingGOPATH).
+
+To check out this repository:
+
+1. Create your own
+   [fork of this repo](https://help.github.com/articles/fork-a-repo/)
+1. Clone it to your machine:
+
+```shell
+mkdir -p ${GOPATH}/src/github.com/kubeflow
+cd ${GOPATH}/src/github.com/kubeflow
+git clone git@github.com:${YOUR_GITHUB_USERNAME}/kfserving.git
+cd serving
+git remote add upstream git@github.com:kubeflow/kfserving.git
+git remote set-url --push upstream no_push
+```
+
+_Adding the `upstream` remote sets you up nicely for regularly
+[syncing your fork](https://help.github.com/articles/syncing-a-fork/)._
+
+Once you reach this point you are ready to do a full build and deploy as
+described below.
+
+## Deploy KFServing
+
+### Check Knative Serving installation
+Once you've [setup your development environment](#prerequisites), you can see things running with:
+
+```console
+kubectl -n knative-serving get pods
```

Comment:
How about adding a `$` or `#` right before this command to distinguish the CLI and output? Ditto for others.